tools/crm_ticket.c

```diff
@@ -791,72 +808,82 @@ main(int argc, char **argv)
-                rc = -EPERM;
+                exit_code = CRM_EX_INSUFFICIENT_PRIV;
                 goto bail;
             }
         }
 
         rc = delete_ticket_state(ticket_id, cib_conn);
+        if (rc != pcmk_ok) {
+            CMD_ERR("Could not clean up ticket: %s", pcmk_strerror(rc));
+        }
+        exit_code = crm_errno2exit(rc);
 
     } else if (modified) {
         if (ticket_id == NULL) {
-            CMD_ERR("Must supply a ticket id with -t\n");
-            rc = -ENXIO;
+            CMD_ERR("Must supply ticket ID with -t");
+            exit_code = CRM_EX_USAGE;
             goto bail;
         }
 
         if (attr_value
             && (attr_name == NULL || strlen(attr_name) == 0)) {
-            CMD_ERR("You need to supply an attribute name with the -S command for -v %s\n", attr_value);
-            rc = -EINVAL;
+            CMD_ERR("Must supply attribute name with -S");
+            exit_code = CRM_EX_USAGE;
             goto bail;
         }
 
         if (attr_name
             && (attr_value == NULL || strlen(attr_value) == 0)) {
-            CMD_ERR("You need to supply a value with the -v option for -S %s\n", attr_name);
-            rc = -EINVAL;
+            CMD_ERR("Must supply attribute value with -v");
+            exit_code = CRM_EX_USAGE;
             goto bail;
         }
 
         if (allow_modification(ticket_id, attr_delete, attr_set) == FALSE) {
-            rc = -EPERM;
+            CMD_ERR("Ticket modification not allowed");
+            exit_code = CRM_EX_INSUFFICIENT_PRIV;
             goto bail;
         }
 
         rc = modify_ticket_state(ticket_id, attr_delete, attr_set, cib_conn, &data_set);
-        
         if (rc != pcmk_ok) {
-            goto bail;
+            CMD_ERR("Could not modify ticket: %s", pcmk_strerror(rc));
         }
+        exit_code = crm_errno2exit(rc);
 
     } else if (ticket_cmd == 'S') {
-        if (ticket_id == NULL) {
-            CMD_ERR("Must supply a ticket id with -t\n");
-            rc = -ENXIO;
+        if (attr_name == NULL || strlen(attr_name) == 0) {
+            // We only get here if ticket_cmd was left as default
+            CMD_ERR("Must supply a command");
+            exit_code = CRM_EX_USAGE;
             goto bail;
         }
 
-        if (attr_name == NULL || strlen(attr_name) == 0) {
-            CMD_ERR("You need to supply a command\n");
-            rc = -EINVAL;
+        if (ticket_id == NULL) {
+            CMD_ERR("Must supply ticket ID with -t");
+            exit_code = CRM_EX_USAGE;
             goto bail;
         }
 
         if (attr_value == NULL || strlen(attr_value) == 0) {
-            CMD_ERR("You need to supply a value with the -v option for -S %s\n", attr_name);
-            rc = -EINVAL;
+            CMD_ERR("Must supply value with -v");
+            exit_code = CRM_EX_USAGE;
             goto bail;
         }
 
+        // Ummm ... should we be doing something here?
```

Comment:
would guess so ;-)

Reply:
According to main(): 631:

```
            case 'S':
                attr_name = optarg;
                if (attr_name && attr_value) {
                    g_hash_table_insert(attr_set, strdup(attr_name), strdup(attr_value));
                    attr_name = NULL;
                    attr_value = NULL;
                    modified++;
                }
                break;
```
All the eligible cases go into the previous block: 807:
```
    } else if (modified) {
...
    }
```

So this block is basically just for reporting specific errors for 'S'.